models/ROMS_rutgers/model_mod.f90

```diff
@@ -0,0 +1,1716 @@
+
+   if (present(last_time) .or. present(all_times)) then
+
+      ! big_integer may overflow a 32bit integer, so declare it 64bit
+      ! and parse it into an integer number of days and seconds, both
+      ! of which can be 32bit. Our set_time, set_date routines need 32bit integers.
+
+      allocate(these_times(dimlen))
+
+      call nc_check(nf90_get_var( ncid, VarID, these_times), &
+             routine, 'get_var '//trim(var_name)//' from '//trim(filename))
+
+      if (present(last_time)) then
+         time_offset = convert_to_time_offset(these_times(dimlen), offset_in_seconds)
+         last_time = base_time + time_offset
+      endif
+
+      if (present(all_times)) then
+         do i=1, dimlen
+            time_offset = convert_to_time_offset(these_times(i), offset_in_seconds)
+            all_times(i) = base_time + time_offset
+         enddo
+      endif
+
+      if (do_output() .and. debug > 0 .and. present(last_time)) then
+         call print_time(last_time, str='last roms time is ',iunit=logfileunit)
+         call print_time(last_time, str='last roms time is ')
+         call print_date(last_time, str='last roms date is ',iunit=logfileunit)
+         call print_date(last_time, str='last roms date is ')
+      endif
+
+      deallocate(these_times)
+
+   endif
+
+   call set_calendar_type(original_calendar_type)
+
+endif
+
+end subroutine get_time_information
+
+
+!-----------------------------------------------------------------------
+! Convert a fractional day to a dart time type
+
+function convert_to_time_offset(offset, offset_in_seconds)
+
+real(digits12), intent(in) :: offset
+logical,        intent(in) :: offset_in_seconds
+type(time_type)            :: convert_to_time_offset
+
+integer(i8) :: big_integer
+integer     :: some_seconds, some_days
+
+if (offset_in_seconds) then
+   big_integer  = int(offset,i8)
+   some_days    = big_integer / (24*60*60)
+   big_integer  = big_integer - (some_days * (24*60*60))
+   some_seconds = int(big_integer,i4)
+else
+   ! offset in fractional days
+   some_days    = int(offset)
+   some_seconds = (offset - some_days) * (24*60*60)
+endif
+
+convert_to_time_offset = set_time(some_seconds, some_days)
+
+end function convert_to_time_offset
+
+
+!-----------------------------------------------------------------------
+! Convert DART time type into a character string with the
+! format of YYYYMMDDhh ... or DDhh
+!
+! time_to_string: Character string containing the time
+! t:              Time
+! interval:       Logical flag describing if the time is to be
+!                 interpreted as a calendar date or a time increment.
+!                 If the flag is merely present, the time is to be
+!                 interpreted as an increment and the format is simply
+!                 DDhh. If the flag is not present, the time is a full
+!                 calendar (Gregorian) date and will be renedered with
+!                 the YYYYMMDDhh format.
+
+function time_to_string(t, interval)
```

Comment:
You can remove this if you remove the adv_to_time optional argument, only place it is used.